app/scripts/modules/core/src/pipeline/config/stages/googleCloudBuild/googleCloudBuildStage.ts

```diff
@@ -12,5 +11,6 @@ Registry.pipeline.registerStage({
   producesArtifacts: true,
   component: GoogleCloudBuildStageConfig,
   executionDetailsSections: [GoogleCloudBuildExecutionDetails, ExecutionDetailsTasks, ExecutionArtifactTab],
-  validators: [{ type: 'requiredField', fieldName: 'account' }],
+  validateFn: validate,
+  artifactExtractor: ExpectedArtifactService.accumulateArtifacts(['buildDefinitionArtifact.artifactId']),
```

Comment:
@ezimanyi do we need an accompanying `artifactRemover` here? I wasn't actually sure why we would need to remove certain artifacts from the context but it seemed to be defined on the stages that included an extractor.

Reply:
The remover handles cleaning up references to artifacts if you remove the artifact from the pipeline so that stages don't have references to artifact id's that don't exist.  The thing to test here would be deleting the expected github artifact in the trigger; ideally this would clear out the artifact in the GCB stage rather than leave an invalid one.

Reply:
Ah, that makes sense, thanks Eric. Added an extractor which won't work yet because the extractors don't seem to be called post-artifacts-rewrite, so I'm going to address that in a followup PR as well as supporting nested references.

Reply:
Followup PR: https://github.com/spinnaker/deck/pull/6939